app/src/main/java/ch/sdp/vibester/activity/PartyRoomActivity.kt

```diff
@@ -1,49 +1,143 @@
 import ch.sdp.vibester.database.Database
+import ch.sdp.vibester.helper.GameManager
 import ch.sdp.vibester.helper.PartyRoom
 import ch.sdp.vibester.user.User
 import com.google.firebase.database.DataSnapshot
 import com.google.firebase.database.DatabaseError
 import com.google.firebase.database.ValueEventListener
+import com.google.gson.Gson
 import dagger.hilt.android.AndroidEntryPoint
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
 import javax.inject.Inject
+import kotlin.properties.Delegates
 
 @AndroidEntryPoint
 class PartyRoomActivity : AppCompatActivity() {
     @Inject
     lateinit var dataGetter: DataGetter
 
+    lateinit var gameManager: GameManager
+
+    lateinit var roomID: String
+
+
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_party_room)
 
-        val roomName = intent.getStringExtra("roomName").toString()
         val createPartyRoom = intent.getBooleanExtra("createRoom", false)
         if(createPartyRoom) {
-            createRoom(roomName)
+            createRoom()
+            setGameManager()
+            dataGetter.updateRoomField(roomID, "gameStarted", false)
         }
         else {
-            fetchData(roomName)
+            this.roomID = intent.getStringExtra("roomID").toString()
+            fetchData(roomID)
         }
+
+        val startGame = findViewById<Button>(R.id.startGame)
+
+        startGame.setOnClickListener {
+//            dataGetter.updateStartGame(roomID, true)
+            dataGetter.updateRoomField(roomID, "gameStarted", true)
+        }
+
+        fetchGameStarted(roomID, this::startGame)
     }
 
-    private fun updateUI(partyRoom: PartyRoom) {
+    private fun updateUI(partyRoom: PartyRoom, roomID: String) {
+        this.roomID = roomID
         findViewById<TextView>(R.id.emails).text = partyRoom.getEmailList().toString()
+        findViewById<TextView>(R.id.roomId).text = roomID
     }
 
-    private fun fetchData(roomName: String) {
-        dataGetter.getRoomData(roomName, this::updateUI)
+    private fun fetchData(roomID: String) {
+        dataGetter.getRoomData(roomID, this::updateUI, this::setSongs)
     }
 
-    private fun createRoom(roomName: String) {
-        dataGetter.createRoom(roomName, this::updateUI)
+    private fun createRoom() {
+        dataGetter.createRoom(this::updateUI)
     }
+
+    private fun fetchGameStarted(roomName: String, callback: (Boolean) -> Unit) {
+        dataGetter.readStartGame(roomName, callback)
+    }
+
+    private fun startGame(gameStarted: Boolean) {
+        if(gameStarted) {
+            launchGame(gameManager)
+        }
+    }
+
+    private fun  setGameManager() {
+        gameManager = GameManager()
+        gameManager.gameSize = 1
+
+        chooseGenre(method = LastfmMethod.BY_ARTIST.method, artist = "Imagine Dragons", mode = R.string.gameGenre_imagine_dragons)
+    }
```

Comment:
Small reminder for the future to make options available to the user when they are setting up a room